File: setup/make-dirs.ts
// tslint:disable:no-console

import * as rimraf from 'rimraf'
import * as fs from 'fs-extra'
import * as path from 'path'

import root from 'zotero-plugin/root'

console.log('make build dirs under', root)
for (let dir of ['build', 'gen', 'gen/typings', 'xpi']) {
  dir = path.join(root, dir)
  console.log('creating', dir)
  rimraf.sync(dir)
  fs.mkdirSync(dir)
}

File: setup/makedirs.py
#!/usr/bin/env python3

import shutil
import os.path

root = os.path.join(os.path.dirname(__file__), '..')

print('make build dirs')
for d in ['build', 'build/content/key-manager', 'build/content/resource/bibtex', 'build/defaults/preferences', 'gen', 'gen/blinkdb', 'gen/preferences', 'gen/typings', 'xpi']:
  print('  creating', d)
  d = os.path.join(root, d)
  if os.path.isdir(d): shutil.rmtree(d)
  os.makedirs(d, exist_ok=True)
